docs/pages/reference/account-kit/wallet-client/functions/grantPermissions.mdx

```diff
@@ -0,0 +1,117 @@
+---
+# This file is autogenerated
+title: grantPermissions
+description: Overview of the grantPermissions method
+slug: wallets/reference/account-kit/wallet-client/functions/grantPermissions
+---
+
+Grants permissions to a smart account by creating a session.
+This allows another key to perform operations on behalf of the account.
+
+## Import
+
+```ts
+import { grantPermissions } from "@account-kit/wallet-client";
+```
+
+## Usage
+
+```ts
+// Create a session key and grant root permissions
+const sessionKey = LocalAccountSigner.generatePrivateKeySigner();
+const account = await client.requestAccount();
+
+const permissions = await client.grantPermissions({
+  account: account.address,
+  expiry: Math.floor(Date.now() / 1000) + 60 * 60, // 1 hour from now
+  key: {
+    publicKey: await sessionKey.getAddress(),
+    type: "secp256k1",
+  },
+  permissions: [{ type: "root" }],
+});
+
+// Use the permissions to prepare a call
+const preparedCalls = await client.prepareCalls({
+  calls: [{ to: zeroAddress, value: "0x0" }],
+  from: account.address,
+  capabilities: {
+    paymasterService: {
+      policyId: "your-paymaster-policy-id",
+    },
+    permissions,
+  },
+});
+
+// Sign with the session key
+const signedCalls = await client.signPreparedCalls(sessionKey, preparedCalls);
```

Comment:
`client.signPreparedCalls` doesn't take a signer – it just uses the signer on the client.
```suggestion
const signedCalls = await signPreparedCalls(sessionKey, preparedCalls);
```

Reply:
the signer on the `client` is the root account signer (which was used to sign the session key). in order for the session key signer to sign calls, we need to either use the `signPreparedCalls` action passing in our session key signer OR create a new client w/ that signer.

Reply:
Ah I see, updating...